src/runner.ts

```diff
@@ -175,9 +178,6 @@ export class Runner {
                 // if not type checking, we don't need to pass in a program object
                 program = undefined;
             }
-        } else if (this.options.typeCheck) {
```

Comment:
all static option validation is done in tslint-cli.ts
runner.ts should only be responsible to catch errors that occur at runtime